hexrd/ui/main_window.py

```diff
@@ -715,9 +718,17 @@ def on_action_edit_euler_angle_convention(self):
         self.update_all()
         self.update_config_gui()
 
+    def active_canvas_changed(self):
+        if hasattr(self, '_apply_drawn_mask_line_picker'):
+            self._apply_drawn_mask_line_picker.canvas_changed(
+                self.ui.image_tab_widget.active_canvas
+            )
```

Comment:
Let's keep the `active_canvas_changed` function, but can we put this code in a different function named `update_drawn_mask_line_picker_canvas()`? Then, `active_canvas_changed()` will call this function.

Then, we should set up the code here so that `active_canvas_changed()` is only actually called when the active canvas changes (there are a few other situations where it is being called right now where the active canvas is not being changed).

Reply:
Changed